{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NetworkError"}],"languages":["swift"],"platforms":["iOS"]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/mgenetwork\/networkerror"],"traits":[{"interfaceLanguage":"swift"}]}],"relationshipsSections":[{"identifiers":["doc:\/\/MGENetwork\/s5ErrorP","doc:\/\/MGENetwork\/s8SendableP"],"kind":"relationships","title":"Conforms To","type":"conformsTo"}],"identifier":{"url":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Error occurring upon network tasks."}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NetworkError"}],"title":"NetworkError","roleHeading":"Enumeration","role":"symbol","symbolKind":"enum","externalID":"s:10MGENetwork12NetworkErrorO","modules":[{"name":"MGENetwork"}],"navigatorTitle":[{"kind":"identifier","text":"NetworkError"}]},"hierarchy":{"paths":[["doc:\/\/MGENetwork\/documentation\/MGENetwork"]]},"topicSections":[{"title":"Enumeration Cases","identifiers":["doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/badRequest","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/encodingFailure","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/forbidden","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/generic(_:)","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/invalidData","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/invalidParameters(parameters:)","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/invalidURL","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/missingInstance","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/notFound","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/unauthorized","doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/unknown"]},{"title":"Instance Properties","identifiers":["doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/message"]},{"title":"Default Implementations","identifiers":["doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/Error-Implementations"],"generated":true}],"references":{"doc://MGENetwork/documentation/MGENetwork":{"role":"collection","title":"MGENetwork","abstract":[],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/generic(_:)":{"role":"symbol","title":"NetworkError.generic(_:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"generic"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"A generic error."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/generic(_:)","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/generic(_:)"},"doc://MGENetwork/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/MGENetwork\/s5ErrorP"},"doc://MGENetwork/documentation/MGENetwork/NetworkError":{"role":"symbol","title":"NetworkError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NetworkError"}],"abstract":[{"type":"text","text":"Error occurring upon network tasks."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"NetworkError"}],"url":"\/documentation\/mgenetwork\/networkerror"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/encodingFailure":{"role":"symbol","title":"NetworkError.encodingFailure","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"encodingFailure"}],"abstract":[{"type":"text","text":"An error occuring while performing encoding operations."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/encodingFailure","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/encodingfailure"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/badRequest":{"role":"symbol","title":"NetworkError.badRequest","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"badRequest"}],"abstract":[{"type":"text","text":"Occurs when "},{"type":"codeVoice","code":"400"},{"type":"text","text":" status code is received from the server."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/badRequest","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/badrequest"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/invalidParameters(parameters:)":{"role":"symbol","title":"NetworkError.invalidParameters(parameters:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidParameters"},{"kind":"text","text":"("},{"kind":"externalParam","text":"parameters"},{"kind":"text","text":": "},{"kind":"keyword","text":"Any"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"Error occuring when building a request with invalid parameter."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/invalidParameters(parameters:)","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/invalidparameters(parameters:)"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/Error-Implementations":{"role":"collectionGroup","title":"Error Implementations","abstract":[],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/Error-Implementations","kind":"article","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/error-implementations"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/unknown":{"role":"symbol","title":"NetworkError.unknown","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"unknown"}],"abstract":[{"type":"text","text":"Used when an unkown error occurs."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/unknown","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/unknown"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/forbidden":{"role":"symbol","title":"NetworkError.forbidden","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"forbidden"}],"abstract":[{"type":"text","text":"Occurs when "},{"type":"codeVoice","code":"403"},{"type":"text","text":" status code is received from the server."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/forbidden","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/forbidden"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/notFound":{"role":"symbol","title":"NetworkError.notFound","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"notFound"}],"abstract":[{"type":"text","text":"Occurs when "},{"type":"codeVoice","code":"404"},{"type":"text","text":" status code is received from the server."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/notFound","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/notfound"},"doc://MGENetwork/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/MGENetwork\/s8SendableP"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/unauthorized":{"role":"symbol","title":"NetworkError.unauthorized","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"unauthorized"}],"abstract":[{"type":"text","text":"Occurs when "},{"type":"codeVoice","code":"401"},{"type":"text","text":" status code is received from the server."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/unauthorized","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/unauthorized"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/message":{"role":"symbol","title":"message","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"message"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"abstract":[{"type":"text","text":"A message for better understanding the error."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/message","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/message"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/invalidURL":{"role":"symbol","title":"NetworkError.invalidURL","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidURL"}],"abstract":[{"type":"text","text":"Error occurring when trying to create an invalid "},{"type":"codeVoice","code":"URL"},{"type":"text","text":" instance."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/invalidURL","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/invalidurl"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/missingInstance":{"role":"symbol","title":"NetworkError.missingInstance","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"missingInstance"}],"abstract":[{"type":"text","text":"The instance does not exist anymore."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/missingInstance","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/missinginstance"},"doc://MGENetwork/documentation/MGENetwork/NetworkError/invalidData":{"role":"symbol","title":"NetworkError.invalidData","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidData"}],"abstract":[{"type":"text","text":"Error occuring when data is invalid."}],"identifier":"doc:\/\/MGENetwork\/documentation\/MGENetwork\/NetworkError\/invalidData","kind":"symbol","type":"topic","url":"\/documentation\/mgenetwork\/networkerror\/invaliddata"}}}